app/js/modules/midi/midi.js

```diff
@@ -74,33 +66,59 @@ define(function(require){
 
     onsuccesscallback:function(access){
 
-            this.m = access;
-            // Things you can do with the MIDIAccess object:
-            var inputs = this.m.inputs; // inputs = MIDIInputMaps, you can retrieve the inputs with iterators
-            var outputs = this.m.outputs; // outputs = MIDIOutputMaps, you can retrieve the outputs with iterators
-            var iteratorInputs = inputs.values() // returns an iterator that loops over all inputs
-            var input = iteratorInputs.next().value // get the first input
-            input.onmidimessage = this.myMIDIMessagehandler.bind(this); // onmidimessage( event ), event.data & event.receivedTime are populated
-            var iteratorOutputs = outputs.values() // returns an iterator that loops over all outputs
-            var output = iteratorOutputs.next().value; // grab first output device
-            output.send( [ 0x90, 0x45, 0x7f ] ); // full velocity note on A4 on channel zero
-            output.send( [ 0x80, 0x45, 0x7f ], window.performance.now() + 1000 ); // full velocity A4 note off in one second.
-
-        },
-
-    myMIDIMessagehandler:function(e){
-
-            if(e.data.length>1){
-                if(e.data[2]>0){
-                    channelMidi.trigger("note:on",0, e.data[1]);
-                }
-                else{
-                    channelMidi.trigger("note:off",0, e.data[1]);
-                }
+        this.access=access;
+        setting.inputs=[];
+        setting.outputs=[];
+        access.inputs.forEach( function( port, key ) {
```

Comment:
Is for each same in this context. I read something that in some cases for with indexes is safer.